File: LeetCode - April/groupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        solution={}
        for i in strs:
            temp="".join(sorted(i))
            if temp in solution:
                solution[temp].append(i)
            else:
                solution[temp]=[i]
        return solution.values()
                

File: LeetCode - April/groupAnagrams.java
import java.util.*;
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        
        HashMap<String,List<String>> solution = new HashMap<>();
        for(int i=0 ; i<strs.length ; i++){
            
            char temp[] = strs[i].toCharArray();
            Arrays.sort(temp);
            String sortedtemp = String.valueOf(temp);
            
            if (!solution.containsKey(sortedtemp)){
                solution.put(sortedtemp,new ArrayList<>());
            }
            solution.get(sortedtemp).add(strs[i]);
        }
        return new ArrayList(solution.values());
    }
}